diff --git a/docroot/sites/all/modules/contrib/dkan_harvest/modules/dkan_harvest_pod/dkan_harvest_pod.migrate.inc b/docroot/sites/all/modules/contrib/dkan_harvest/modules/dkan_harvest_pod/dkan_harvest_pod.migrate.inc
index 011db9f..eb558d3 100644
--- a/docroot/sites/all/modules/contrib/dkan_harvest/modules/dkan_harvest_pod/dkan_harvest_pod.migrate.inc
+++ b/docroot/sites/all/modules/contrib/dkan_harvest/modules/dkan_harvest_pod/dkan_harvest_pod.migrate.inc
@@ -22,41 +22,14 @@ function dkan_harvest_pod_migrate_api() {
   return $api;
 }
 
-class PODJsonHarvestList extends MigrateList {
-
-  protected $files;
-
-  /**
-   * Builds the list of files paths inside DKAN_HARVEST_CACHE_DIR.
-   */
-  public function __construct(HarvestSource $dkan_harvest_source) {
-    parent::__construct();
-    if (isset($dkan_harvest_source)) {
-      // Build list of items based off sources definitions.
-      $this->files = file_scan_directory($dkan_harvest_source->getCacheDir(), '(.*)');
-    }
-  }
-
-  /**
-   * Implements MigrateList::getIdList().
-   */
-  public function getIdList() {
-    return array_keys($this->files);
-  }
-
-  /**
-   * Implements MigrateList::__toString().
-   */
-  public function __toString() {
-    return '';
-  }
+class PODJsonHarvestList extends HarvestList {
 
   /**
    * Implements MigrateList::computeCount().
    */
   public function computeCount() {
     $count = count($this->getIdList());
-    foreach ($this->files as $file->uri) {
+    foreach ($this->files as $file) {
       if (file_exists($file->uri)) {
         $data = json_decode(file_get_contents($file->uri));
         $count = $count + count($data->distribution);
@@ -129,16 +102,17 @@ class PODJsonHarvest extends MigrateDataJsonDatasetBase {
    */
   public function __construct($arguments) {
     parent::__construct($arguments);
-    $fields = $this->getDataJsonDatasetFields();
+    if (isset($arguments['dkan_harvest_source'])) {
     $dkan_harvest_source = $arguments['dkan_harvest_source'];
     $this->dkan_harvest_source = $dkan_harvest_source;
-    $podJsonHarvestList = new PODJsonHarvestList($dkan_harvest_source);
-    $podJsonHarvestItem = new PODJsonHarvestItem($dkan_harvest_source);
+    $fields = $this->getDataJsonDatasetFields();
+
     $this->source = new MigrateSourceList(
-      $podJsonHarvestList,
-      $podJsonHarvestItem,
+      new PODJsonHarvestList($dkan_harvest_source->getCacheDir()),
+      new PODJsonHarvestItem($dkan_harvest_source),
       $fields
     );
+    }
   }
 
   /**
